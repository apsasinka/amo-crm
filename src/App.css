.app {
  font-family: Arial, sans-serif;
  padding: 20px;
}

h1 {
  margin-bottom: 20px;
}

.grid-container {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 15px;
  position: relative;
  margin: 20px 0;
}

.lead-card {
  border: 1px solid #ccc;
  padding: 10px;
  border-radius: 5px;
  background-color: #f9f9f9;
  cursor: pointer;
  transition: all 0.3s ease;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

.lead-card.selected {
  grid-column: span 1;
  grid-row: span 2;
  z-index: 1;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
}

.lead-details {
  display: none;
  margin-top: 10px;
  background-color: #e9e9e9;
  padding: 10px;
  border-radius: 5px;
}

.lead-card.selected .lead-details {
  display: block;
}

.loading-spinner {
  text-align: center;
  font-size: 14px;
}

svg {
  margin-left: 5px;
}