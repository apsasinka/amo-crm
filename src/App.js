// import { useState, useEffect } from "react";
// import "./App.css";
// import Users from "./users/Users";

// function App() {
//   const [data, setData] = useState(null);
//   const [toggle, setToggle] = useState(false);
//   const [value, setValue] = useState("");

//   const handleToggle = () => {
//     setToggle((prev) => !prev);
//   };

//   useEffect(() => {
//     setTimeout(() => setData({}), 0);
//   }, []);

//   return (
//     <div className="app">
//       <h1 data-testid="input-value">{value}</h1>
//       {toggle && <div data-testid="toggle-elem">toggle</div>}
//       {data && <div style={{ color: "red" }}>Data</div>}
//       <button onClick={handleToggle} data-testid="toggle-btn">
//         Hy
//       </button>
//       <input
//         onChange={(e) => setValue(e.target.value)}
//         placeholder="write text..."
//       />
//       <Users />
//     </div>
//   );
// }

import React from "react";
import { Link } from "react-router-dom";

const App = () => {
  return (
    <>
      <Link data-testid="main-link" to="/" style={{ margin: "3em" }}>
        main
      </Link>
      <Link data-testid="about-link" to="/about">
        about
      </Link>
      <Link data-testid="users-link" to="/users">
        Users
      </Link>
    </>
  );
};

export default App;
