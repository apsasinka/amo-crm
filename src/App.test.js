import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";

describe("Test APP", () => {
  test("renders learn react link", () => {
    render(<App />);
    const leadsTextElem = screen.getByTestId("input-value");
    const btn = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/write text/i);

    expect(leadsTextElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test("HZ", async () => {
    render(<App />);
    // const missingElem = screen.queryByText(/test/i);
    const dataElem = await screen.findByText(/data/i);

    // expect(missingElem).toBeNull();
    expect(dataElem).toHaveStyle({ color: "red" });
  });

  test("Test for event listener", () => {
    render(<App />);

    const btn = screen.getByTestId("toggle-btn"); //когда мы получаем объект, то можем заметить, что у него нет каких-либо методов из формы. Для использования методов нам необходим специальный объект fireEvent

    expect(screen.queryByTestId("toggle-elem")).toBeNull();

    fireEvent.click(btn); //первым аргументом передаем необходимый элемент, а вторым может передать дополнительные опции
    expect(screen.queryByTestId("toggle-elem")).toBeInTheDocument();
  });

  test("Input event", () => {
    render(<App />);

    const input = screen.getByPlaceholderText(/write text.../i);
    expect(screen.queryByTestId("input-value")).toContainHTML(""); //при помощи данного метода мы можем установить соответствие текста внутри HTML

    fireEvent.input(input, {
      target: { value: "123123" },
    });

    expect(screen.queryByTestId("input-value")).toContainHTML("123123");
  });
});
