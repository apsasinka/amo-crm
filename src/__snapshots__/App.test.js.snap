// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test APP renders learn react link 1`] = `
<input
  placeholder="write text..."
/>
`;
